src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs

```diff
@@ -89,6 +94,144 @@ public async Task<CitizenDocumentUploadResponse> UploadAttachmentWithHttpMessage
+            var responseSasUri = await client.StorageAccessWithHttpMessagesAsync(
+                                   subscriptionId: new Guid(this.options.SubscriptionId),
+                                   documentName: storageDocName).ConfigureAwait(false);
+
+            if (responseSasUri.Response.StatusCode != System.Net.HttpStatusCode.OK)
+            {
+                throw new Exception("You don't have permission to upload");
+            }
+
+            var sasTokenUri = new Uri(responseSasUri.Body);
+
+            var containerAddress = new Uri($"{sasTokenUri.Scheme}://{sasTokenUri.Host}/{sasTokenUri.AbsolutePath.Split('/')[1]}");
+
+            CloudBlobContainer container = new CloudBlobContainer(
+                containerAddress,
+                new StorageCredentials(sasTokenUri.Query));
+
+            var uploadResponse = await UploadDocumentToAzureStorage(document, storageDocName, container, 5000000).ConfigureAwait(false);
+
+            if (uploadResponse.StatusCode != System.Net.HttpStatusCode.OK)
+            {
+                throw new Exception("Upload Failed");
+            }
+
+            citizenDocumentUploadRequestModel.Id = documentId;
+            citizenDocumentUploadRequestModel.DocumentUrl = $"{containerAddress.ToString()}/{storageDocName}";
+            citizenDocumentUploadRequestModel.Status = "Completed";
+
+            var updateResponse = await client.UpdateDataToDbWithHttpMessagesAsync(
+                                 subscriptionId: new Guid(this.options.SubscriptionId),
+                                 request: citizenDocumentUploadRequestModel).ConfigureAwait(false);
+            switch (updateResponse.Response.StatusCode)
+            {
+                case System.Net.HttpStatusCode.OK:
+                    return (CitizenDocumentUploadResponse)updateResponse.Body;
+
+                case System.Net.HttpStatusCode.Unauthorized:
+                    throw new CitizenDocumentsException("Unauthorized", updateResponse.Body as string);
+
+                default:
+                    throw new CitizenDocumentsException("Invalid configuration provided to access Citizen Document service", updateResponse.Body as string);
+            }
+        }
+
+        private static async Task<UploadResponseModel> UploadDocumentToAzureStorage(Stream document, string documentName, CloudBlobContainer container, int size = 5000000)
+        {
+            if (documentName == null)
+            {
+                throw new ArgumentNullException(nameof(documentName));
+            }
+
+            CloudBlockBlob blob = container.GetBlockBlobReference(documentName);
+            try
+            {
+                int bytesRead;
+                int blockNumber = 0;
+                List<string> blockList = new List<string>();
+                do
+                {
+                    Console.WriteLine(blockNumber);
```

Comment:
What if I create windows or mobile application and I don't have console?

Reply:
Removed console.writeline